chore: read and use nonce from network instead from the local cache when sending tx

saledjenic opened the pull request:
Reading the Nonce from the local cache may be incorrect if the tx is made out of the Status app or if Status app sends a tx prepared by the dapp (via WalletConnect). A submitted tx with a wrong Nonce results in a failing tx, that's why we need to read the Nonce from the network.

Affected areas: wallet + other parts of the app that send a transaction

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | a806d8c6 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4489/1/) | 2023-12-19 12:46:58 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231219-124336-a806d8-pr4489.zip) |
| :heavy_check_mark: | a806d8c6 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4489/1/) | 2023-12-19 12:48:34 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231219-124331-a806d8-pr4489.zip) |
| :heavy_check_mark: | a806d8c6 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4489/1/) | 2023-12-19 12:49:01 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231219-124331-a806d8-pr4489.aar) |
| :heavy_multiplication_x: | a806d8c6 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4489/1/) | 2023-12-19 13:17:57 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4489/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 36915259 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4489/2/) | 2023-12-19 13:41:55 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231219-133826-369152-pr4489.zip) |
| :heavy_check_mark: | 36915259 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4489/2/) | 2023-12-19 13:44:07 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231219-133826-369152-pr4489.aar) |
| :heavy_multiplication_x: | 36915259 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4489/2/) | 2023-12-19 14:06:22 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4489/2//consoleText) |
| :heavy_check_mark: | 36915259 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4489/3/) | 2023-12-20 08:15:50 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4489/3//consoleText) |
| :heavy_check_mark: | 36915259 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4489/8/) | 2023-12-20 11:20:03 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231220-111610-369152-pr4489.zip) |

saledjenic committed:
chore: read and use nonce from network instead from the local cache when sending tx

Reading the Nonce from the local cache may be incorrect if the tx is made out of the Status app or
if Status app sends a tx prepared by the dapp (via WalletConnect). A submitted tx with a wrong Nonce
results in a failing tx, that's why we need to read the Nonce from the network.

alaibe reviewed (approved):
Code LGTM, Just curious if we are not missing an edge case
@cammellos ?

saledjenic commented:
@cammellos I missed the PR and accidentally merged this instead of another PR, if you have any suggestions/comments let me know and I will apply them in a new PR.
